opentelemetry-sdk/tests/test_util.py

```diff
@@ -0,0 +1,162 @@
+
+        for idx in range(list_len):
+            self.assertEqual(blist[idx], TestBoundedList.base[idx])
+
+        # sequence too big
+        with self.assertRaises(ValueError):
+            BoundedList.from_seq(list_len / 2, TestBoundedList.base)
+
+    def test_append_no_drop(self):
+        # create empty list
+        list_len = len(TestBoundedList.base)
+        blist = BoundedList(list_len)
+        self.assertEqual(len(blist), 0)
+
+        # fill list
+        for idx in TestBoundedList.base:
+            blist.append(idx)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, 0)
+
+        for idx in range(list_len):
+            self.assertEqual(blist[idx], TestBoundedList.base[idx])
+
+        # test __iter__ in BoundedList
+        idx = 0
+        for val in blist:
+            self.assertEqual(val, TestBoundedList.base[idx])
+            idx += 1
+
+    def test_append_drop(self):
+        list_len = len(TestBoundedList.base)
+        # create full BoundedList
+        blist = BoundedList.from_seq(list_len, TestBoundedList.base)
+
+        # try to append more items
+        for idx in range(8):
+            # should drop the element without raising exceptions
+            blist.append(idx)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, 8)
+
+    def test_extend_no_drop(self):
+        # create empty list
+        list_len = len(TestBoundedList.base)
+        blist = BoundedList(list_len)
+        self.assertEqual(len(blist), 0)
+
+        # fill list
+        blist.extend(TestBoundedList.base)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, 0)
+
+        for idx in range(list_len):
+            self.assertEqual(blist[idx], TestBoundedList.base[idx])
+
+        # test __iter__ in BoundedList
+        idx = 0
+        for val in blist:
+            self.assertEqual(val, TestBoundedList.base[idx])
+            idx += 1
+
+    def test_extend_drop(self):
+        list_len = len(TestBoundedList.base)
+        # create full BoundedList
+        blist = BoundedList.from_seq(list_len, TestBoundedList.base)
+        other_list = [13, 37, 51, 91]
+
+        # try to extend with more elements
+        blist.extend(other_list)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, len(other_list))
+
+
+class TestBoundedDict(unittest.TestCase):
+    base = collections.OrderedDict(
+        [
+            ("name", "Firulais"),
+            ("age", 7),
+            ("weight", 13),
+            ("vaccinated", True),
+        ]
+    )
+
+    def test_negative_maxlen(self):
+        with self.assertRaises(ValueError):
+            BoundedDict(-1)
+
+    def test_from_map(self):
+        dic_len = len(TestBoundedDict.base)
+        bdict = BoundedDict.from_map(dic_len, TestBoundedDict.base)
+
+        self.assertEqual(len(bdict), dic_len)
+
+        for key in TestBoundedDict.base:
```

Comment:
You could also test that all keys are yielded by the iterator like asserting that `len(tuple(bdict)) == len(bdict))`. Same for BoundedList.